camviewer_ui_impl.py

```diff
@@ -337,218 +328,124 @@ def __init__(self, app, cwd, instrument, camera, cameraPv,
+    self.ui.checkBoxProjAutoRange.stateChanged.connect(self.onCheckProjUpdate)
     
-    self.connect(self.ui.horizontalSliderRangeMin , QtCore.SIGNAL("valueChanged(int)"), self.onSliderRangeMinChanged )
-    self.connect(self.ui.horizontalSliderRangeMax, QtCore.SIGNAL("valueChanged(int)"), self.onSliderRangeMaxChanged)
-    self.connect(self.ui.lineEditRangeMin , QtCore.SIGNAL("returnPressed()"), self.onRangeMinTextEnter )
-    self.connect(self.ui.lineEditRangeMax, QtCore.SIGNAL("returnPressed()"), self.onRangeMaxTextEnter)    
-
-    self.connect(self.ui.horizontalSliderLens , QtCore.SIGNAL("sliderReleased()"), self.onSliderLensReleased )
-    self.connect(self.ui.horizontalSliderLens , QtCore.SIGNAL("valueChanged(int)"), self.onSliderLensChanged )
-    self.connect(self.ui.lineEditLens , QtCore.SIGNAL("returnPressed()"), self.onLensEnter )
-
-    self.connect(self.ui.singleframe, QtCore.SIGNAL("toggled(bool)"), self.onCheckDisplayUpdate)
-    self.connect(self.ui.grayScale,   QtCore.SIGNAL("stateChanged(int)"), self.onCheckGrayUpdate)
-    self.connect(self.ui.rem_avg,     QtCore.SIGNAL("toggled(bool)"), self.onCheckDisplayUpdate)
-    self.connect(self.ui.local_avg,   QtCore.SIGNAL("toggled(bool)"), self.onCheckDisplayUpdate)
-    self.connect(self.ui.remote_average, QtCore.SIGNAL("returnPressed()"), self.onRemAvgEnter )
+    self.ui.horizontalSliderRangeMin.valueChanged.connect(self.onSliderRangeMinChanged)
+    self.ui.horizontalSliderRangeMax.valueChanged.connect(self.onSliderRangeMaxChanged)
+    self.ui.lineEditRangeMin.returnPressed.connect(self.onRangeMinTextEnter)
+    self.ui.lineEditRangeMax.returnPressed.connect(self.onRangeMaxTextEnter)    
+
+    self.ui.horizontalSliderLens.sliderReleased.connect(self.onSliderLensReleased)
+    self.ui.horizontalSliderLens.valueChanged.connect(self.onSliderLensChanged)
+    self.ui.lineEditLens.returnPressed.connect(self.onLensEnter)
+
+    self.ui.singleframe.toggled.connect(self.onCheckDisplayUpdate)
+    self.ui.grayScale.stateChanged.connect(self.onCheckGrayUpdate)
+    self.ui.rem_avg.toggled.connect(self.onCheckDisplayUpdate)
+    self.ui.local_avg.toggled.connect(self.onCheckDisplayUpdate)
+    self.ui.remote_average.returnPressed.connect(self.onRemAvgEnter)
     
-    self.connect(self.ui.comboBoxColor, QtCore.SIGNAL("currentIndexChanged(int)"), self.onComboBoxColorIndexChanged)
+    self.ui.comboBoxColor.currentIndexChanged.connect(self.onComboBoxColorIndexChanged)
     self.hot() # default option
 
     for i in range(4):
-      self.connect(self.ui.xmark[i], QtCore.SIGNAL("returnPressed()"),
-                   functools.partial(self.onMarkerTextEnter, i))
-      self.connect(self.ui.ymark[i], QtCore.SIGNAL("returnPressed()"),
-                   functools.partial(self.onMarkerTextEnter, i))
+      self.ui.xmark[i].returnPressed.connect(functools.partial(self.onMarkerTextEnter, i))
```

Comment:
It's worth noting that pyqt has an issue with partial methods in that it won't automatically disconnect signals connected with them at clean-up.

I think this instance would run for the life of the program, but if there are other such instances you may want to look into alternatives

Here's what we added to typhos to make it easy-to-use: https://github.com/pcdshub/typhos/blob/14af093e703baccfe4e18c2eef5d394d9213c8fe/typhos/utils.py#L390-L472

Reply:
Yeah, I think I'm using partials all over the place.  I'll take a look at that link...

Reply:
Taking a quick look, all of the signal owners belong to either the main gui or a dialog... and all of the methods belong to the main implementation class.  *Everything* lives for the life of the program... so I think I'm OK here.

Reply:
(I reuse dialogs and don't create and destroy them as needed.)